dask/optimization.py

```diff
@@ -808,3 +894,47 @@ def key_split(s):
             return result
     except Exception:
         return 'Other'
+
+
+class SubgraphCallable(object):
+    """Create a callable object from a dask graph.
+
+    Parameters
+    ----------
+    dsk : dict
+        A dask graph
+    outkey : hashable
+        The output key from the graph
+    inkeys : list
+        A list of keys to be used as arguments to the callable.
+    name : str, optional
+        The name to use for the function.
+    """
+    __slots__ = ('dsk', 'outkey', 'inkeys', 'name')
+
+    def __init__(self, dsk, outkey, inkeys, name='subgraph_callable'):
+        self.dsk = dsk
+        self.outkey = outkey
+        self.inkeys = inkeys
+        self.name = name
+
+    def __repr__(self):
+        return self.name
+
+    def __eq__(self, other):
+        return (type(self) is type(other) and
+                self.dsk == other.dsk and
+                self.outkey == other.outkey and
+                set(self.inkeys) == set(other.inkeys) and
+                self.name == other.name)
```

Comment:
Where do we use eq here?  Tests?  Do we also need to implement `__hash__` if we define `__eq__`?

Reply:
Tests. We don't need to define `__hash__`, as the objects don't need to be hashable (and hashing them may be expensive anyway).

I do need to add a `__ne__` implementation for python 2.7 though.